recipe/activate-mdt.sh

```diff
@@ -1,20 +1,41 @@
-if [ -z "$MACOSX_DEPLOYMENT_TARGET" ]; then
-    export MACOSX_DEPLOYMENT_TARGET="@MACOSX_DEPLOYMENT_TARGET@"
-else
-    export CONDA_SYSROOT_@PLATFORM@_BACKUP_MACOSX_DEPLOYMENT_TARGET="${MACOSX_DEPLOYMENT_TARGET}"
+#!/usr/bin/env bash
+
+if [[ "${CONDA_BUILD:-0}" == "1" ]]; then
+    if [ -n "$MACOSX_DEPLOYMENT_TARGET" ]; then
+        export CONDA_SYSROOT_@PLATFORM@_BACKUP_MACOSX_DEPLOYMENT_TARGET="${MACOSX_DEPLOYMENT_TARGET}"
+    fi
     export MACOSX_DEPLOYMENT_TARGET="@MACOSX_DEPLOYMENT_TARGET@"
-fi
 
-if [ -z "${CMAKE_ARGS}" ]; then
-    export CMAKE_ARGS="-DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET}"
-else
-    export CONDA_SYSROOT_@PLATFORM@_BACKUP_CMAKE_ARGS="${CMAKE_ARGS}"
-    export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET}"
-fi
+    # From a conda-build environment, we will rely on OSX_SDK_DIR being set in order to set CONDA_BUILD_SYSROOT to a
+    # SDK matching the MACOSX_DEPLOYMENT_TARGET. If not set, we will use the SDKROOT or CONDA_BUILD_SYSROOT, hoping
+    # that they are set to a SDK matching the MACOSX_DEPLOYMENT_TARGET.
```

Comment:
```suggestion
    # We rely on OSX_SDK_DIR being set in order to set CONDA_BUILD_SYSROOT to a SDK matching the
    # MACOSX_DEPLOYMENT_TARGET. If not set, we will use in priority CONDA_BUILD_SYSROOT, then 
    # SDKROOT, hoping that they are set to a SDK matching the MACOSX_DEPLOYMENT_TARGET.
    # In the end, both CONDA_BUILD_SYSROOT and SDKROOT are set to match.
```

Reply:
We could also echo a warning if MACOSX_DEPLOYMENT_TARGET is not found in CONDA_BUILD_SYSROOT.